### PR TITLE
Update CMake settings to match recent package changes

### PR DESCRIPTION
**Pull request by stmontgomery**
A few fixes to the CMake and package build settings:

- Enable Swift 6 language mode, to match the package which now uses this language mode due to the change which bumped `swift-tools-version` to 6.0.
- Add SourceLocation macro-related source files, which were missing.
- Add fake `visionOS 99.0` OS version to our `_distantFuture` availability definition.
- Clean up experimental/upcoming features which are no longer needed.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by stmontgomery** [comment]
@swift-ci please test

**Comment by stmontgomery** [comment]
@compnerd @jeffdav FYI about these adjustments to the CMake rules.

File: cmake/modules/shared/CompilerSettings.cmake [hunk 1]
```diff
@@ -8,14 +8,14 @@
 
 # Settings intended to be applied to every Swift target in this package.
 # Analogous to project-level build settings in an Xcode project.
+add_compile_options(
+  "SHELL:$<$<COMPILE_LANGUAGE:Swift>:-swift-version 6>")
```
Comment (compnerd) [inline cmake/modules/shared/CompilerSettings.cmake:12]: I think that this is better spelt as the following at the top level CMakeLists.txt:

```cmake
set(Swift_LANGUAGE_VERSION 6)
```

**Review by compnerd — APPROVED** [review]

File: cmake/modules/shared/CompilerSettings.cmake (hunk 1 above)
Comment (stmontgomery) [reply cmake/modules/shared/CompilerSettings.cmake]: Looks like it's spelled **CMAKE**_Swift_LANGUAGE_VERSION, but that does work!

**Comment by stmontgomery** [comment]
@swift-ci please test

**Comment by stmontgomery** [comment]
@swift-ci please test

**Review by briancroom — APPROVED** [review]

**Pull request merged**